File: frontend/src/service/patient-service.ts
import api, { IPageable } from './api';

export interface IPatient {
    id?: number
    active: boolean
    socialSecurityNumber: string
    firstname: string
    lastname: string
    email: string | null
}

const getPatients = async (page: number, size: number = 50): Promise<IPageable<IPatient>> => {
  const response = await api.get(
    '/patients',
    {
      params: {
        page,
        size
      }
    }
  )
  return Promise.resolve(response.data)
}

const savePatient = async (patient: IPatient): Promise<IPatient> => {
  const response = await api.post(
    '/patients',
    { ...patient }
  )
  return response.data
}

const updatePatient = async (patient: IPatient): Promise<IPatient> => {
  const response = await api.put(
    '/patients',
    { ...patient }
  )
  return response.data
}

const patientService = {
  getPatients,
  savePatient,
  updatePatient
}

export default patientService

File: backend/src/main/java/io/p3admin/service/PatientService.java
package io.p3admin.service;

import io.p3admin.model.domain.Patient;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PatientService {
    Page<Patient> getPatients(Pageable pageable);

    Patient savePatient(Patient patient);

    Patient updatePatient(Patient patient);
}
